.github/workflows/node.js.yml

```diff
@@ -53,24 +57,33 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-lintcache-
 
+      - name: Cache node modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            node_modules
+            packages/*/node_modules
+          key: node-modules-${{ hashFiles('package-lock.json')}}
+
       - name: Install dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
         run: npm ci --no-audit
 
       - name: Build
-        run: npm run build
-
+        run: VITE_CORE_API_URL=http://localhost:10000/jsapi npm run build
```

Comment:
If running `npm run preview:app` for e2e, this might not be necessary. The Vite preview server runs w/ the proxies which default to `localhost:10000`

Reply:
This wasn't working for me without specifying the core api URL when I was testing. I'll try it one more time to see if it breaks.

Reply:
Pretty sure removing this causes the tests to just stall: https://github.com/deephaven/web-client-ui/actions/runs/3704468319/jobs/6277124073

I'll add it back in a bit.

Reply:
You can try with `VITE_PROXY_URL=http://localhost:10000` when you run preview, but this should be fine too.

Actually without the proxy, things like notebooks might be broken since the notebook URL is proxied to port 10000 only if the proxy url is specified and not running w/ Docker. `preview` runs in production mode, so there's no proxy url by default

Reply:
The e2e tests check notebook saving/opening functionality, so notebooks seem to be working